ScoutSuite/providers/gcp/services/computeengine.py

```diff
@@ -188,3 +197,39 @@ def parse_firewalls(self, firewall, params):
                                     firewall_dict[direction_string][rule['IPProtocol']] = ['0-65535']
 
         self.firewalls[firewall_dict['id']] = firewall_dict
+
+    def _get_project_id(self, instance):
+        return instance['selfLink'].split('/')[-5]
+
+    def _metadata_to_dict(self, metadata):
+        return dict((item['key'], item['value']) for item in metadata['items']) if 'items' in metadata else {}
+
+    def _is_block_project_ssh_keys_enabled(self, instance):
+        metadata = self._metadata_to_dict(instance['metadata'])
+        return metadata.get('block-project-ssh-keys') == 'true'
+
+    def _get_common_instance_metadata_dict(self, project_id):
+        computeengine_client = gcp_connect_service(service='computeengine')
+        request = computeengine_client.projects().get(project=project_id)
```

Comment:
I think you should be able to get an instance of the client by hsing

```py
params['api_client'].compute_engine()
```
Or something like that (not sure about the `compute_engine` part.

Reply:
Done!

Reply:
Yep, you don't need to instantiate a client from the service object as it's passed down to it.